libsigopt/compute/log_likelihood.py

```diff
@@ -152,10 +163,8 @@ def compute_grad_log_likelihood(self, include_nonzero_correction=INCLUDE_NONZERO
       dKKinvy_Pb = numpy.dot(dK, Kinvy_Pb)
       grad_log_marginal[k] = -numpy.dot(Kinvy_Pb, dKKinvy_Pb)
       grad_log_marginal[k] += numpy.trace(scipy.linalg.cho_solve(K_chol, dK, overwrite_b=True))
-      if not self.gp.has_zero_mean and include_nonzero_correction:
-        P = self.gp.P
-        KinvP = self.gp.K_inv_P
-        PKP_chol = self.gp.PKP_chol
+      if gp_params is not None:
+        P, KinvP, PKP_chol = gp_params
```

Comment:
I see what you are doing here but it's better to keep it as it was
```
if not self.gp.has_zero_mean and include_nonzero_correction:
```
that's an additional term that we need to add to the `grad_log_marginal` only if that if is true. Changing it to `gp_params` will make this code harder to read than already is.

Reply:
I also think the ordering is different, in 149: 
```
gp_params = self.gp.P, self.gp.K_inv_P, self.gp.K_inv_demeaned_y
```
but here we want the `PKP_chol`

Reply:
Yeah sorry there are some issues here I need to clean up.

The reason for assigning these to `gp_params` is because mypy want's to be certain that `self.gp.P`, `self.gp.K_inv_P` and `self.gp.PKP_chol` are not `None` before they are used. Instead placing `assert ... is not None` in the for loop (inside the removed conditional statement), it is simpler (and more performant) to move `if not self.gp.has_zero_mean ...` and the aforementioned assertions outside the for loop. This way we only need to check for the presence of `gp_params` to know that none of the params are `None` and that the condition is true.

Reply:
can you call the variable `gp_terms_for_non_zero_mean` then?